dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java

```diff
@@ -74,21 +92,89 @@ public void shouldNotSaveNewRule() throws DotSecurityException, DotDataException
         this.addPermission(role, host, false);
 
         try {
-            APILocator.getRulesAPI().saveRule(rule, user, false);
+            rulesAPI.saveRule(rule, user, false);
             throw new AssertionError("DotSecurityException expected");
         } catch(DotSecurityException e) {
             //expected//
         }
 
-        final List<Rule> rules = APILocator.getRulesAPI().getAllRulesByParent(host, APILocator.systemUser(), true);
+        final List<Rule> rules = rulesAPI.getAllRulesByParent(host, systemUser, true);
         assertTrue(rules.isEmpty());
     }
 
+    /**
+     * Method to Test: {@link RulesAPIImpl#deleteRule(Rule, User, boolean)}
+     * Case: Happy path
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void testDeleteRuleShouldPass() throws DotSecurityException, DotDataException {
+        long currentTime = System.currentTimeMillis();
+        final Rule rule  = new RuleDataGen().name("testDeleteRule" + currentTime).nextPersisted();
+        final RuleAction ruleAction = new RuleActionDataGen().ruleId(rule.getId()).nextPersisted();
+        final ConditionGroup conditionGroup = new ConditionGroupDataGen().ruleId(rule.getId()).nextPersisted();
+        final Condition condition = new ConditionDataGen().groupId(conditionGroup.getId()).nextPersisted();
+
+        rulesAPI.deleteRule(rule, systemUser, false);
+
+        rulesCache.removeRule(rule);
+        rulesCache.removeAction(ruleAction);
+        rulesCache.removeCondition(condition);
+        rulesCache.removeConditionGroup(conditionGroup);
+
+        assertNull(rulesAPI.getRuleById(rule.getId(), systemUser, false));
+        assertNull(rulesAPI.getRuleActionById(ruleAction.getId(), systemUser, false));
+        assertNull(rulesAPI.getConditionById(condition.getId(), systemUser, false));
+        assertNull(rulesAPI.getConditionGroupById(conditionGroup.getId(), systemUser, false));
+
+
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#deleteRule(Rule, User, boolean)}
+     * When: a rule is deleted, the delete operation over its dependencies should not fail because of a dirty cache
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void testDeleteRuleShouldNotFailWhenCacheIsDirty() throws DotSecurityException, DotDataException {
+        long currentTime = System.currentTimeMillis();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'currentTime' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4979626)